nodeup/pkg/model/context.go

```diff
@@ -183,3 +183,30 @@ func (c *NodeupModelContext) UsesCNI() bool {
 	}
 	return true
 }
+
+// UseSecureKubelet checks if the kubelet api should be protected by a client certificate. Note: the settings are be
+// in one of three section, master specific kubelet, cluster wide kubelet or the InstanceGroup. Though arguably is
+// doesn't make much sense to unset this on a per InstanceGroup level, but hey :)
```

Comment:
I agree that it doesn't really make a difference, because it doesn't make any sense to mix this setting, but I feel like the logic should be:

if instancegroup anonymousauth != nil, return that value
if master and masterkubelet anonymousauth != nil, return that value
if kubelet anonymousauth != nil, return that value

i.e. most specific to least specific, and you can set anonymousauth explicitly to true on the instancegroup and override (though this is silly in practice)